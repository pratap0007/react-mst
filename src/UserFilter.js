import React from "react";
import { StoreContext } from "./store/store";

export const UserFilter = () => {
  const store = React.useContext(StoreContext);
  const filter = [];


 

  return (
    <div>
      <h3>Filter: </h3>
      <button id="bx" type='submit' onClick={console.warn('button fired')}>
          test

      </button>

      <div
        onChange={(e) => {
          if (
            e.target.value === "FY" ||
            e.target.value === "SY" ||
            e.target.value === "TY"
          ) {
            if (e.target.checked === true) {
              filter.push(e.target.value);
            } else {
              var index = filter.indexOf(e.target.value);
              while (index !== -1) {
                filter.splice(index, 1);
                index = filter.indexOf(e.target.value);
              }
            }
            store.setFilter(filter);
          }
        }}
      >
        <input type="checkbox" id="c1" defaultChecked={false} value="FY" /> FY
        <input type="checkbox" id="c2" defaultChecked={false} value="SY" /> SY
        <input type="checkbox" id="c3" defaultChecked={false} value="TY" /> TY
      </div>
      <div
        onChange={(e) => {
          if (e.target.value === "Name") {
            store.setSort("name");
          } else if (e.target.value === "Age") {
            store.setSort("age");
          } else if (e.target.value === "No") {
            store.setSort("");
          }
        }}
      >
        <h3>Sort By:</h3>
        <input type="radio" name="options" value="Name" /> Name
        <input type="radio" name="options" value="Age" /> Age
        <input type="radio" name="options" value="No" defaultChecked /> No Sort
      </div>
    </div>
  );
};
