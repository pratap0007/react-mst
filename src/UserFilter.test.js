import React from "react";
import renderer from "react-test-renderer";
import fs from "fs";
import { Store } from "./store/userlist";
import { StoreContext } from "./store/store";
import "./App.css";
import { UserFilter } from "./UserFilter.js";
import { mount, shallow } from "enzyme";

const userFetcher = () =>
  Promise.resolve(JSON.parse(fs.readFileSync("./src/store/users.json")));

const store = Store.create({}, { fetch: userFetcher });

const StoreProvider = ({ children }) => {
  return (
    <StoreContext.Provider value={store}>{children}</StoreContext.Provider>
  );
};

describe("<UserFilter />", () => {
  it("should render correctly", () => {
    const component = renderer.create(<UserFilter />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

it("should render banner text correctly with given strings", () => {
  // const component = shallow(
  //   <StoreProvider>
  //     <UserFilter />
  //   </StoreProvider>
  // );
  const component = mount(
    <StoreProvider>
      <UserFilter />
    </StoreProvider>
  );
const mockCallBack = jest.fn();

  // component.find('input[name="options"]').at(1).simulate("click");
  // var checkbox = () => component.find('input[type="radio"]').at(1);
  component.find('#bx').simulate('click')
  // component
  //   .find('input[type="checkbox"]')
  //   .at(0)
  //   .simulate('change', { target: { defaultChecked: true } });
  // expect(component).toMatchSnapshot();
  console.log(store.count);
  const list = store.userList;
  console.log(list[0].name);
});

// const component = mount(
//     <StoreProvider>
//       <UserFilter />
//     </StoreProvider>
//   );
//   expect(component).toMatchSnapshot();
//   // component.find("checkbox#c1").simulate('change', {target: {checked: true}});
//   var checkbox = component.find("#checkbox").eq('c1');
//   checkbox.simulate("change", { target: { checked: true } });
//   component.unmount();
