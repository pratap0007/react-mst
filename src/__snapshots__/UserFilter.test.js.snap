// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserFilter /> should render correctly 1`] = `
<div>
  <h3>
    Filter: 
  </h3>
  <button
    id="bx"
    type="submit"
  >
    test
  </button>
  <div
    onChange={[Function]}
  >
    <input
      defaultChecked={false}
      id="c1"
      type="checkbox"
      value="FY"
    />
     FY
    <input
      defaultChecked={false}
      id="c2"
      type="checkbox"
      value="SY"
    />
     SY
    <input
      defaultChecked={false}
      id="c3"
      type="checkbox"
      value="TY"
    />
     TY
  </div>
  <div
    onChange={[Function]}
  >
    <h3>
      Sort By:
    </h3>
    <input
      name="options"
      type="radio"
      value="Name"
    />
     Name
    <input
      name="options"
      type="radio"
      value="Age"
    />
     Age
    <input
      defaultChecked={true}
      name="options"
      type="radio"
      value="No"
    />
     No Sort
  </div>
</div>
`;
